### PR TITLE
Group Resource

### PR DESCRIPTION
**Pull request by Stanislav2017**
Signed-off-by: Stanislav2017 stanislavchichek@gmail.com

### Description of your changes

This adds Groups managed resource to provider-gitlab.

Fixes https://github.com/crossplane-contrib/provider-gitlab/issues/4

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

manually

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Review by janwillies — CHANGES_REQUESTED** [review]
Hey @Stanislav2017, thanks for the contribution! Some small comments, but looking good already!

**Comment by janwillies** [comment]
Please also squash your commits into a single one, you can read about it here: https://www.internalpointers.com/post/squash-commits-into-one-git

File: pkg/controller/groups/group.go [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+	}
+	if !clients.IsIntEqualToIntPtr(p.ExtraSharedRunnersMinutesLimit, g.ExtraSharedRunnersMinutesLimit) {
+		return false
+	}
+	return true
+}
+
+// LateInitialize fills the empty fields in the group spec with the
+// values seen in gitlab.Group.
+func LateInitialize(in *v1alpha1.GroupParameters, group *gitlab.Group) { // nolint:gocyclo
```
[the remark this answers is not in the document]
Comment (Stanislav2017) [reply pkg/controller/groups/group.go:216]: I think no. I will rename this method to lateInitialize

File: pkg/controller/groups/group.go [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+		return errors.New(errNotGroup)
+	}
+
+	cr.Status.SetConditions(xpv1.Deleting())
+
+	_, err := e.client.DeleteGroup(meta.GetExternalName(cr), gitlab.WithContext(ctx))
+	return errors.Wrap(err, errDeleteFailed)
+}
+
+// IsGroupUpToDate checks whether there is a change in any of the modifiable fields.
```
[the remark this answers is not in the document]
Comment (Stanislav2017) [reply pkg/controller/groups/group.go:167]: I think no. I will rename this method to isGroupUpToDate

**Review by janwillies — APPROVED** [review]
this looks very good, thanks a lot @Stanislav2017! FWIW it has been reviewed in https://github.com/crossplane-contrib/provider-gitlab/pull/18 and in a now-deleted repo as well.

**Pull request merged**